### ISSUE TITLE
Feature request: support IMap.getValues on set of keys

### ISSUE THREAD

**Issue by dmitrymz** [hazelcast/hazelcast#8870]
Currently IMap supports values(Predicate predicate), and it would be convenient to add a method similar to executeOnKeys but for filtering with predicate, ie 

values(Set<K> keys, Predicate predicate);
entrySet(Set<K> keys, Predicate predicate);

looking at source code it shouldn't be difficult as instead of calling invokeQueryAllPartitions the method will use getPartitionsForKeys(), then invokeOnPartitions()

This opens new possibilities of cross map joins

**Comment by jerrinot** [comment]
hi @dmitrymz, thanks for you idea. Wouldn't something like this work for you?
```java
Predicate p = Predicates.and(
                       originalPredicate,
                       Predicates.in("__key", keys)
                     );
```

**Comment by dmitrymz** [comment]
Hi, @jerrinot. thanks for reply. my only concern with example above that it doesn't avoid full table scan. when we feed that predicate p into existing values(Predicate) method it asks all partitions first, then does the "in" clause which is too late.

**Comment by jerrinot** [comment]
I understand. Perhaps we could be somewhat smarter internally and optimize queries like this. 
This way we could keep support your use-case without introducing a new API.

**Comment by dmitrymz** [comment]
yes, it could be a special predicate just taking set of keys - that's fine. internally your code can do optimizations. Basically using this pattern of predefined predicate object lots of apis can be collapsed into one. then internally optimizations will do the magic.

would be great to see it live one day :)
